x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -632,6 +654,196 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     });
   }
 
+  public async collectMultiNamespaceReferences(
+    objects: SavedObjectsCollectMultiNamespaceReferencesObject[],
+    options: SavedObjectsCollectMultiNamespaceReferencesOptions = {}
+  ): Promise<SavedObjectsCollectMultiNamespaceReferencesResponse> {
+    const result = await this.collectMultiNamespaceReferencesInternal(objects, options);
+    return { objects: result.objects };
+  }
+
+  public async updateObjectsSpaces(
+    objects: SavedObjectsUpdateObjectsSpacesObject[],
+    spacesToAdd: string[],
+    spacesToRemove: string[],
+    options: SavedObjectsUpdateObjectsSpacesOptions
+  ) {
+    const { namespace, includeReferences, excludeTags } = options;
+    const currentSpaceId = SavedObjectsUtils.namespaceIdToString(namespace); // We need this whether the Spaces plugin is enabled or not.
+
+    const uniqueSpaces = this.getUniqueSpaces(currentSpaceId, ...spacesToAdd, ...spacesToRemove);
+    let objectsToUpdate: SavedObjectsUpdateObjectsSpacesObject[];
+    let authorizationTypeMap: Map<string, EnsureAuthorizedTypeResult>;
+    if (includeReferences) {
+      const collectResult = await this.collectMultiNamespaceReferencesInternal(objects, {
```

Comment:
Fair enough! It also occurred to me today that this will likely simplify the conflict resolution process for when you are attempting to share an object to a space where a matching Legacy URL alias already exists with the same ID (to prevent the "alias conflict" scenario)